indico/web/client/js/react/components/category_navigator/view.jsx

```diff
@@ -0,0 +1,350 @@
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Dialog from 'indico/react/components/Dialog';
+import {Translate, PluralTranslate, Param} from 'indico/react/i18n';
+import 'indico/custom_elements/ind_with_tooltip';
+import 'indico/custom_elements/ind_with_toggletip';
+
+import './style.scss';
+
+const debounceTimer = Symbol('debounceTimer');
+const debounceTimeout = 500;
+
+function LoadingView({loading, children}) {
+  return (
+    <div className="category-list-container">
+      <p className="categories-loading" aria-live="polite">
+        {loading ? <Translate as="span">Please wait.</Translate> : null}
+      </p>
+      {loading ? null : children}
+    </div>
+  );
+}
+
+LoadingView.propTypes = {
+  loading: PropTypes.bool,
+  children: PropTypes.oneOfType([PropTypes.element, PropTypes.arrayOf(PropTypes.element)]),
+};
+
+function ErrorView({error}) {
+  return (
+    <p className="category-load-error" role="alert">
+      {error ? <Translate>Categories could not be loaded.</Translate> : null}
+    </p>
+  );
+}
+
+ErrorView.propTypes = {
+  error: PropTypes.object,
+};
+
+function CategorySearchView({searchFieldRef, onSearch}) {
+  // XXX: We're not using a controlled component for the input. Because a lot of the operations surrounding it are async,
+  // we don't want to have to deal with races. Instead, we're going to reset the input manually when the time is right.
+
+  function handleSearch(ev) {
+    clearTimeout(ev.target[debounceTimer]);
+    ev.target[debounceTimer] = setTimeout(() => {
+      onSearch(ev.target.value);
+    }, debounceTimeout);
+  }
+
+  return (
+    <form onSubmit={e => e.preventDefault()}>
+      <label className="category-search-field">
+        <span>
+          <Translate>Find a category by keyword:</Translate>
+        </span>
+        <input
+          ref={searchFieldRef}
+          type="search"
+          placeholder={Translate.string('Search all categories...')}
+          aria-describedby="category-search-info"
+          onChange={handleSearch}
+        />
+      </label>
+      <p className="category-search-info">
+        <Translate>Search results will update as you type.</Translate>
+      </p>
+    </form>
+  );
+}
+
+CategorySearchView.propTypes = {
+  searchFieldRef: PropTypes.object,
+  onSearch: PropTypes.func.isRequired,
+};
+
+function BreadcrumbView({path, onChangeCategory}) {
+  if (!path?.length) {
+    return null;
+  }
+
+  return (
+    <div className="breadcrumbs">
+      <span>{Translate.string('in')}</span>
+      {path.map(category => (
+        <button
+          type="button"
+          key={category.id}
+          onClick={() => onChangeCategory(category.id)}
+          aria-label={Translate.string('View the {title} category', {title: category.title})}
```

Comment:
```suggestion
          aria-label={Translate.string('View the "{title}" category', {title: category.title})}
```

Depending on the category title not quoting it may look strange

Reply:
It's an aria label, so it doesn't "look". :) Screen readers will not do anything special with quotes, afaik.

Reply:
But let me double-check that just in case.

Reply:
ah, missed that. nvm then :)

Reply:
Ok, it really makes no difference. But since it makes no difference, I'll add the quote just in case there are ATs that do something with them.